python/lsst/pipe/base/connections.py

```diff
@@ -309,9 +309,7 @@ def __call__(cls, *, config: PipelineTaskConfig | None = None) -> PipelineTaskCo
         # Extract the template names that were defined in the config instance
         # by looping over the keys of the defaultTemplates dict specified at
         # class declaration time.
-        templateValues = {
-            name: getattr(config.connections, name) for name in getattr(cls, "defaultTemplates").keys()
-        }
+        templateValues = {name: getattr(config.connections, name) for name in cls.defaultTemplates}
```

Comment:
mypy is getting very confused by this line. It's convinced that `cls` can't have `defaultTemplates` but if it didn't have it then the getattr would complain.